sample-service/src/main/java/uk/ac/ox/ndph/mts/sample_service/security/authroisation/AuthorisationService.java

```diff
@@ -1,42 +1,74 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import java.util.List;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /**
  * The service which performs the authorisation flow
  */
 @Service
 public class AuthorisationService {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthorisationService.class);
+
+    private final SecurityContextComponent securityContextComponent;
+
+    private final PractitionerServiceClient practitionerServiceClient;
+    private final RoleServiceClient roleServiceClientImpl;
+
+    @Autowired
+    public AuthorisationService(final SecurityContextComponent securityContextComponent,
+                                final PractitionerServiceClient practitionerServiceClient,
+                                final RoleServiceClient roleServiceClientImpl) {
+        this.securityContextComponent = securityContextComponent;
+        this.practitionerServiceClient = practitionerServiceClient;
+        this.roleServiceClientImpl = roleServiceClientImpl;
+    }
+
     /**
      * Authorise request
      * @return true if authorised
      */
-    @SuppressWarnings("squid:S2589") //allow using method that always return true, currently a stub algorithm
-    public boolean authorise(String requiredPermission) {
-        //currently
-        String userId = "stubUserId";
+    public boolean authorise(String requiredPermission)  {
 
-        //get practitioner assignment role
-        List<String> participantRoles = getParticipantRoles(userId);
+        try {
+            //Get Azure Active Directory user object id
+            String userId = getUserIdFromContext();
 
-        //get permissions for the the practitioner assignment roles
-        List<String> permissions = participantRoles.stream().flatMap(role ->
-                getRolePermissions(role).stream()).collect(Collectors.toList());
+            //get practitioner assignment role
+            AssignmentRoleDTO[] participantRoles = practitionerServiceClient.getUserAssignmentRoles(userId);
 
-        //validate the required permission is present
-        if (!permissions.contains(requiredPermission)) {
-            return false;
-        }
+            if (participantRoles == null || participantRoles.length == 0) {
+                LOGGER.info("User with id {} has no assignment Roles and therefore is unauthorised.", userId);
+                return false;
+            }
 
-        //we will also update the practitioner roles to be only those who have the permissions to perform the action
-        //currently just a stub to show we filtered them out
+            //get permissions for the the practitioner assignment roles
+            //and filter assignment roles to be only those which have the required permission in them
+            var rolesWithPermission = Stream.of(participantRoles)
+                    .map(role -> roleServiceClientImpl.getRolesById(role.getRoleId()))
```

Comment:
Can we batch this call to the roles service? Otherwise could have performance implications. May need a new endpoint on the roles service.